oak_crypto/src/noise_nk/noise.rs

```diff
@@ -0,0 +1,252 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! This was copied from Chromium's third_party/cloud_authenticator, which has
+//! compatible copyright and ownership (Apache 2.0, Google).
+
+use alloc::vec::Vec;
+
+use crate::noise_nk::{error::Error, rustcrypto};
+
+const SYMMETRIC_KEY_SIZE: usize = 32;
+pub const NONCE_LEN: usize = 12;
+
+#[derive(Debug, PartialEq)]
+pub enum HandshakeType {
+    Nk, // https://noiseexplorer.com/patterns/NK/
+}
+
+// Helper to generate 2 keys.
+fn hkdf2(
+    ck: &[u8; SYMMETRIC_KEY_SIZE],
+    ikm: &[u8],
+) -> ([u8; SYMMETRIC_KEY_SIZE], [u8; SYMMETRIC_KEY_SIZE]) {
+    let mut output = [0; rustcrypto::SHA256_OUTPUT_LEN * 2];
+    // unwrap: only fails if the output size is too large, but the output
+    // size is small and fixed here.
+    rustcrypto::hkdf_sha256(ikm, ck, &[], &mut output).unwrap();
+    let key1: [u8; SYMMETRIC_KEY_SIZE] = output[..SYMMETRIC_KEY_SIZE].try_into().unwrap();
+    let key2: [u8; SYMMETRIC_KEY_SIZE] = output[SYMMETRIC_KEY_SIZE..].try_into().unwrap();
+    (key1, key2)
+}
+
+#[derive(Debug, PartialEq)]
```

Comment:
See previous comment about removing the `Debug` from structs that contain secrets

Reply:
Removed.